### PR TITLE
Upgrade to the platform generator 0.0.37

### PR DESCRIPTION
**Pull request by aloubyansky**
This release disables the alignment of member own constraints if no other member includes constraints that create a conflict. (This also reduces the platform generation time). The previous behavior can be enabled by adding `<alignOwnConstraints>true</alignOwnConstraints>` element to a member's config.

For example, Camel Quarkus imports the Azure SDK BOM and no other member includes these constraints. This release will not attempt to align them by default.

FYI @ppalaga @jamesnetherton @vkasala

### CODE REVIEW

File: generated-platform-project/quarkus-camel/bom/pom.xml [hunk 1]
```diff
@@ -647,17 +647,17 @@
       <dependency>
         <groupId>io.dropwizard.metrics</groupId>
         <artifactId>metrics-healthchecks</artifactId>
-        <version>4.1.19</version>
+        <version>4.0.1</version>
```
Comment (aloubyansky) [inline generated-platform-project/quarkus-camel/bom/pom.xml:650]: This change (and other dropwizard, org.graalvm.js and apache.curator) indicate that your BOM includes other artifacts that originate from different releases of the same project. I.e. they were previously aligned by the platform generator but now they are not.

Comment (ppalaga) [reply generated-platform-project/quarkus-camel/bom/pom.xml:650]: This is a bit strange.

For dropwizard and Camel Quarkus 2.4.0, I'd expect the version 4.1.25, defined in `org.apache.camel:camel-dependencies:3.12.0`

Similarly `org.graalvm.js` should be at `21.2.0` see https://github.com/apache/camel-quarkus/blob/2.4.0/poms/bom/pom.xml#L6084 and https://github.com/apache/camel-quarkus/blob/2.4.0/pom.xml#L83

Comment (aloubyansky) [reply generated-platform-project/quarkus-camel/bom/pom.xml:650]: You have two artifacts coming from `org.graalvm.js`: `js` one is `21.2.0` but the `js-scriptengine` is `20.0.0`.
As to the dropwizard, you should check but that's what they actually are. An easy way to verify it is to download https://repo1.maven.org/maven2/org/apache/camel/quarkus/camel-quarkus-bom/2.4.0/camel-quarkus-bom-2.4.0.pom as `pom.xml` and run `mvn help:effective-pom > effective-pom.xml`. You'll see the flattened version of your BOM with all the actual constraints.

Comment (ppalaga) [reply generated-platform-project/quarkus-camel/bom/pom.xml:650]: Going to check that now

Comment (ppalaga) [reply generated-platform-project/quarkus-camel/bom/pom.xml:650]: You are right, the BOM generator works as designed, we need to be more careful what we manage in our BOM. I filed https://github.com/apache/camel-quarkus/issues/3302 for that. 
These insights are super useful for us - thanks a lot for them!

**Review by ppalaga — APPROVED** [review]

**Comment by ppalaga** [comment]
BTW, in the prod platform 2.2.x, I'll be for having `<alignOwnConstraints>true</alignOwnConstraints>` and keeping the enforcedDependencies

```
                            <enforcedDependencies>
                                <!-- Workaround for https://issues.redhat.com/browse/QUARKUS-1379 -->
                                <dependency>com.azure:azure-storage-blob-batch:12.9.1</dependency>
                                <dependency>com.azure:azure-storage-file-share:12.9.1</dependency>
                                <dependency>com.azure:azure-storage-queue:12.9.1</dependency>
                            </enforcedDependencies>
```

because we do not want to rebuild Camel Quarkus.

**Comment by ppalaga** [comment]
Would you please ping me when you have backported this to the prod branch?

**Comment by gsmet** [comment]
I wonder if we should merge this until things are fixed in Camel Quarkus?

**Comment by ppalaga** [comment]
> I wonder if we should merge this until things are fixed in Camel Quarkus?

https://github.com/apache/camel-quarkus/pull/3303 contains the minimal set of fixes and I hope it will be merged tomorrow. So a fixed Camel Quarkus BOM should be ready for Platform 2.5.0.Final

**Comment by gsmet** [comment]
OK, no objections from me in this case.

**Comment by ppalaga** [comment]
https://github.com/apache/camel-quarkus/pull/3303 is merged now.

**Pull request merged**